packages/babel-plugin-transform-modules-amd/test/fixtures/amd/export-from/output.js

```diff
@@ -4,14 +4,15 @@ define(["exports", "foo"], function (_exports, _foo) {
   Object.defineProperty(_exports, "__esModule", {
     value: true
   });
-  Object.keys(_foo).forEach(function (key) {
+
+  function _exportFromThis(key) {
     if (key === "default" || key === "__esModule") return;
-    if (key in _exports && _exports[key] === _foo[key]) return;
+    if (key in _exports && _exports[key] === this[key]) return;
     Object.defineProperty(_exports, key, {
       enumerable: true,
-      get: function () {
-        return _foo[key];
-      }
+      get: () => this[key]
     });
-  });
+  }
+
+  Object.keys(_foo).forEach(_exportFromThis, _foo);
```

Comment:
Is it okay to capture `_foo` here? In the original code, the getter evaluated `_foo[key]`, whereas the new code captures the value of `_foo` on this line and doesn't read the variable afterwards. This should be fine as long as `_foo` is not reassigned, but I'm not 100% sure whether that's guaranteed.

Reply:
`_foo` is a function parameter injected by Babel and not modifiable by user code, so it's ok.